test-suite/jvm/src/test/scala/org/scalajs/testsuite/utils/Platform.scala

```diff
@@ -15,4 +15,18 @@ object Platform {
     if (v.startsWith("1.")) Integer.parseInt(v.drop(2).takeWhile(_.isDigit))
     else throw new Exception("Unknown java.version format")
   }
+
+  def executingInRhino: Boolean = false
+  def executingInNodeJS: Boolean = false
+  def executingInPhantomJS: Boolean = false
+  def hasTypedArrays: Boolean = false
+
+  def isInFastOpt: Boolean = false
+  def isInFullOpt: Boolean = false
+  def isInStrongMode: Boolean = false
+  def isInProductionMode: Boolean = false
+
+  def hasCompliantAsInstanceOfs: Boolean = true
+  def hasCompliantModule: Boolean = true
+  def hasStrictFloats: Boolean = true
```

Comment:
Not yet, but they will allow me to port the jasmine tests that use these conditions.